File: 1886-determine-whether-matrix-can-be-obtained-by-rotation/1886-determine-whether-matrix-can-be-obtained-by-rotation.js
/**
 * @param {number[][]} mat
 * @param {number[][]} target
 * @return {boolean}
 */
var findRotation = function(mat, target) {
    const rotations = [true, true, true, true]
    const width = mat.length
    for (let r=0; r < width; r++) {
        for (let c=0; c < width; c++) {
            if (mat[r][c] !== target[r][c]) rotations[0] = false
            if (mat[r][c] !== target[width-c-1][r]) rotations[1] = false
            if (mat[r][c] !== target[width-r-1][width-c-1]) rotations[2] = false
            if (mat[r][c] !== target[c][width-r-1]) rotations[3] = false
        }
    }
    return rotations.some(rot => rot === true)
};

File: 1886-determine-whether-matrix-can-be-obtained-by-rotation/1886-determine-whether-matrix-can-be-obtained-by-rotation.py
class Solution:
    def findRotation(self, mat: List[List[int]], target: List[List[int]]) -> bool:
        rot = [1] * 4
        w = len(mat)
        for r in range(0, w):
            for c in range(0, w):
                if mat[r][c] != target[r][c]:
                    rot[0] = 0
                if mat[r][c] != target[w-c-1][r]:
                    rot[1] = 0
                if mat[r][c] != target[w-r-1][w-c-1]:
                    rot[2] = 0
                if mat[r][c] != target[c][w-r-1]:
                    rot[3] = 0                
        return sum(rot) > 0